### PR TITLE
Fix the removed counts in SIS logging

### PR DESCRIPTION
**Pull request by PasiSa**
SIS logging incorrectly counted earlier removed students as removed on every updated. Now it counts only those participants who were not removed earlier.

# Description

**What?**

[ANSWER HERE]

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

Fixes #<issue number>

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: course/models.py [hunk 1]
```diff
@@ -964,11 +964,13 @@ def enroll_from_sis(self) -> Tuple[int, int]:
...
-            qs = Enrollment.objects.filter(user_profile__in=to_remove, course_instance=self)
-            qs.update(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
+            qs = (Enrollment.objects
+                  .filter(user_profile__in=to_remove, course_instance=self)
+                  .exclude(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
+            )
             for e in qs:
                 invalidate_content(Enrollment, e)
-                delcount += 1
+            delcount = qs.update(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
```
Comment (Mankro) [inline course/models.py:973]: Would it still make more sense to remove the students before invalidating the cache? `qs.update()` used to come before `invalidate_content()`.

**Review by Mankro — CHANGES_REQUESTED** [review]
Good. I just think it is best to run the operation and then clear the cache, not the other way around.

File: course/models.py (hunk 1 above)
Comment (PasiSa) [reply course/models.py:973]: I tried that first, but it seems that after the update operation the qs becomes empty here. Apparently the update can affect the qs contents if the same attribute that is updated is also used for filtering.

Comment (Mankro) [reply course/models.py:973]: Right, I didn't think of that. At least, one possibility is to copy the original list before running `update()`. 
```
qs_before_update = list(qs)
delcount = qs.update(status=...)
for e in qs_before_update:
    invalidate_content(...)
```

Do you prefer to change this or merge the pull request without further changes?

Comment (PasiSa) [reply course/models.py:973]: I don't have strong preference, so either option is fine with me, but is the order between invalidate and update that important? Compared to having a new variable for a temporary list?

**Review by Mankro — APPROVED** [review]

File: course/models.py (hunk 1 above)
Comment (Mankro) [reply course/models.py:973]: It is not a big deal. We can keep it like you originally did.

**Pull request merged**